styx-api-service/src/test/java/com/spotify/styx/api/WorkflowResourceTest.java

```diff
@@ -816,14 +833,202 @@ public void shouldReturnWorkflows() throws Exception {
+            createWorkflowWithTypeAndTime("id2", "remote-foo", Instant.parse(queryThresholdBefore)),
+            createWorkflowWithTime("id3", Instant.parse(queryThresholdAfter))
+            )
+    );
+
+    Response<ByteString> response = awaitResponse(
+        serviceHelper.request("GET", path("?deployment_type=remote-foo&deployment_time_before=" + queryThreshold)));
+
+    assertThat(response, hasStatus(withCode(Status.OK)));
+    assertJson(response, "[*]", hasSize(1));
+    assertJson(response, "[0].component_id", is("id2"));
+  }
+
+  @Test
+  public void shouldReturnFilteredDeploymentTypeTimeAfterWorkflow() throws Exception {
+    sinceVersion(Api.Version.V3);
+
+    var queryThreshold = "2022-01-01T10:15:30.00Z";
+    var queryThresholdBefore = "2022-01-01T10:15:29.00Z";
+    var queryThresholdAfter = "2022-01-01T10:15:31.00Z";
+
+    when(storage.workflows()).thenReturn(
+        buildWorkflowMap(
+            createWorkflowWithType("id1", "remote-foo"),
+            createWorkflowWithTime("id2", Instant.parse(queryThresholdBefore)),
+            createWorkflowWithTypeAndTime("id3", "remote-foo", Instant.parse(queryThresholdAfter))
+        )
+    );
+
+    Response<ByteString> response = awaitResponse(
+        serviceHelper.request("GET", path("?deployment_time_after=" + queryThreshold)));
+
+    assertThat(response, hasStatus(withCode(Status.OK)));
+    assertJson(response, "[*]", hasSize(1));
+    assertJson(response, "[0].component_id", is("id3"));
+  }
+
+  @Test
+  public void shouldReturnFilteredDeploymentTypeTimeBeforeAndAfterWorkflow() throws Exception {
+    sinceVersion(Api.Version.V3);
+    var deploymentTimeAfter = "2022-01-01T10:15:28.00Z";
+    var deploymentTimeBefore = "2022-01-01T10:15:32.00Z";
+    var queryThresholdBefore = "2022-01-01T10:15:29.00Z";
+    var queryThresholdAfter = "2022-01-01T10:15:31.00Z";
+
+    when(storage.workflows()).thenReturn(
+        buildWorkflowMap(
+            createWorkflowWithType("id1", "remote-foo"),
+            createWorkflowWithTypeAndTime("id2", "remote-foo", Instant.parse(queryThresholdBefore)),
+            createWorkflowWithTime("id3", Instant.parse(queryThresholdAfter))
+        )
+    );
+
+    Response<ByteString> response = awaitResponse(
+        serviceHelper.request("GET", path("?deployment_type=remote-foo&deployment_time_after="
+            + deploymentTimeAfter
+            + "&deployment_time_before="
+            + deploymentTimeBefore)));
+
+    assertThat(response, hasStatus(withCode(Status.OK)));
+    assertJson(response, "[*]", hasSize(1));
+    assertJson(response, "[0].component_id", is("id2"));
+  }
+
   @Test
   public void shouldFailedToReturnWorkflows() throws Exception {
     sinceVersion(Api.Version.V3);
 
     when(storage.workflows()).thenThrow(new IOException());
 
     Response<ByteString> response = awaitResponse(
-        serviceHelper.request("GET", path("")));
+        serviceHelper.request("GET", path("?deployment_type=")));
```

Comment:
```suggestion
        serviceHelper.request("GET", path("")));
```
It seems that we can revert this change